test/lib/code_corps/github/event/installation/repos_test.exs

```diff
@@ -16,20 +16,38 @@ defmodule CodeCorps.GitHub.Event.Installation.ReposTest do
 
   alias CodeCorps.GitHub.Adapters.GithubRepo, as: GithubRepoAdapter
 
-  @access_token "v1.1f699f1069f60xxx"
-  @expires_at Timex.now() |> Timex.shift(hours: 1) |> DateTime.to_iso8601()
-  @access_token_create_response %{"token" => @access_token, "expires_at" => @expires_at}
+  defmodule BadRepoRequest do
+    alias CodeCorps.GitHubCase
 
-  @installation_repositories load_endpoint_fixture("installation_repositories")
-  @forbidden load_endpoint_fixture("forbidden")
+    def request(:get, "https://api.github.com/installation/repositories", _, _, _) do
+      body = GitHubCase.Helpers.load_endpoint_fixture("forbidden")
+      {:error, CodeCorps.GitHub.APIError.new({404, %{"message" => body}})}
+    end
+    def request(method, endpoint, headers, body, options) do
+      GitHubCase.SuccessAPI.request(method, endpoint, headers, body, options)
+    end
+  end
+
+  defmodule InvalidRepoRequest do
+    alias CodeCorps.GitHubCase
+
+    def request(:get, "https://api.github.com/installation/repositories", _, _, _) do
+      payload =
+        "installation_repositories"
+        |> GitHubCase.Helpers.load_endpoint_fixture
+        |> Map.put("repositories", [%{}])
+      {:ok, payload}
+    end
+    def request(method, endpoint, headers, body, options) do
+      GitHubCase.SuccessAPI.request(method, endpoint, headers, body, options)
+    end
+  end
```

Comment:
This is an example of using mock modules to replace the api layer temporarily for specific tests.

The code being tested here makes two requests and the tests specify what happens when one of them fails in a specific way. 

We've defined modules for those two failure cases where the method signature which ought to fail, fails in that specific way, while all other method signatures _delegate_ to the default `GitHubCase.SuccessAPI` module.